test/smoke/src/areas/positron/dataexplorer/veryLargeDataFrame.test.ts

```diff
@@ -0,0 +1,142 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import { expect } from '@playwright/test';
+import { Application, downloadFileFromS3, Logger, PositronPythonFixtures, PositronRFixtures, S3FileDownloadOptions } from '../../../../../automation';
+import { installAllHandlers } from '../../../utils';
+import { join } from 'path';
+import { fail } from 'assert';
+
+/*
+ * Data explorer test suite for large data frames
+ */
+export function setup(logger: Logger) {
+
+	// AWS Configuration
+	const region = "us-west-2";
+	const bucketName = "positron-qa-data-files";
+	const objectKey = "largeParquet.parquet";
+
+	describe('Data Explorer - Very Large Data Frame', () => {
+
+		// Shared before/after handling
+		installAllHandlers(logger);
+
+		before(async function () {
+
+			const localFilePath = join(this.app.workspacePathOrFolder, "data-files", objectKey);
+
+			const downloadOptions: S3FileDownloadOptions = {
+				region: region,
+				bucketName: bucketName,
+				key: objectKey,
+				localFilePath: localFilePath
+			};
+			await downloadFileFromS3(downloadOptions);
```

Comment:
Would it make sense to simulate a large random dataset instead of downloading a large one?

Reply:
Great question.  Brian requested that we hit all possible data types with this test, which is not as easy with a randomized data frame.  I am actually not sure how to get some of the data types without using a parquet file.  We do have a utility for creating test data frames, but it is in Rust and I was shying away from getting it to work in CI.  Plus, the utility execution time is greater than the S3 download time, I am pretty sure.

Reply:
We could also download a 100 or 1000 rows data frame and replicate it 10000 times too, so we easily get all data types. The code in R and Pandas/Polars should be pretty simple to do this.

Reply:
That is very true.  Maybe I could make one of each type and see how they perform in comparison? I will add second cases that do what you suggest and do a little comparing.

Reply:
(leaving out Polars for now while we get issues resolved but will add later)

Reply:
This actually produced interesting results.  It appears that with [multiplyParquet.py](https://github.com/posit-dev/qa-example-content/blob/main/workspaces/performance/multiplyParquet.py) & [multiplyParquet.r](https://github.com/posit-dev/qa-example-content/blob/main/workspaces/performance/multiplyParquet.r), the opening of the files is pretty significantly faster:

Opening large unique parquet took 2921.2224170000045 milliseconds (pandas)
Opening large duplicated parquet took 2059.595208999999 milliseconds (pandas)

Opening large unique parquet took 6403.574999999997 milliseconds (R)
Opening large duplicated parquet took 3825.589749999999 milliseconds (R)

Reply:
I think keeping both tests makes sense to me.  What do you think, @dfalbel ?

Reply:
My main concern with adding S3 downloaded files is that it makes harder for us to run the tests locally if needed. I'm also not sure they are testing different things. For sure, opening a large file stresses `pd.read_parquet()` or `arrow::read_parquet()` more, but AFAICT, once the data frames are in-memory, there's no difference in behavior for the data explorer, so these are essentially testing/forcing the same codepaths.

Reply:
Yeah, I am going to add in logic to not run them locally (Pete suggested that) but I am only timing the data frame open (not the file read).  Is it surprising that unique data opens more slowly than duplicated blocks?

Reply:
Ok, local skip logic is in there.  It will only do the duplicated frames locally.

Reply:
There's some lazy loading mechanisms in `read_parquet()` that will delay loading the entire data frame until the data is actually entirely requested. What's probably happening here is that the data frame is that data explorer is finally loading it. But could be other things, like, interactions with the variables pane, etc.

Reply:
These are still 2 different scenarios.  It makes sense to keep the quicker test running locally, and keeping the S3 test only in the nightly CI run.  It's a valid user scenario and if they start seeing it get slower, we should at least be able to catch that it's happening, regardless of the root cause.